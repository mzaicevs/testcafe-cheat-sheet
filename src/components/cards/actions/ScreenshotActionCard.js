import React from "react";
import { actions_takeScreenshot } from "../../../snippets";
import CodeCard from "../../../base_components/CodeCard";

const ScreenshotActionCard = () => (
  <CodeCard
    cardTitle={actions_takeScreenshot.title}
    fullDocLink={actions_takeScreenshot.fullDocLink}
    codeSnippet={{
      code: actions_takeScreenshot.code,
      language: actions_takeScreenshot.language,
      options: actions_takeScreenshot.options,
      example: actions_takeScreenshot.example
    }}
    description={actions_takeScreenshot.description}
  />
);

export default ScreenshotActionCard;
