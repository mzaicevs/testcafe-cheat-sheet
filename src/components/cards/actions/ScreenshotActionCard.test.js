import React from "react";
import { shallow } from "enzyme";
import ScreenshotActionCard from "./ScreenshotActionCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("ScreenshotActionCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<ScreenshotActionCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
