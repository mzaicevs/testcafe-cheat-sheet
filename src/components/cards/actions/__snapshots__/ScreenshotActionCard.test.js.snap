// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenshotActionCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Take Screenshot - Action"
  codeSnippet={
    Object {
      "code": Array [
        "t.takeScreenshot( [path] )",
      ],
      "example": "import { Selector } from 'testcafe';

  fixture \`My fixture\`
      .page \`http://devexpress.github.io/testcafe/example/\`;
  
  test('Take a screenshot of a fieldset', async t => {
      await t
          .typeText('#developer-name', 'Tony Stark')
          .click('#submit-button')
          .takeScreenshot('my-fixture/thank-you-page.png')
          .takeElementScreenshot(Selector('fieldset').nth(1), 'my-fixture/important-features.png')
          .takeElementScreenshot('#avatar', {
            includeMargins: true,
            crop: {
                top: -100,
                left: 10,
                bottom: 30,
                right: 200
            }
        });
  });",
      "language": "javascript",
      "options": "{
    scrollTargetX: Number,
    scrollTargetY: Number,
    includeMargins: Boolean,
    includeBorders: Boolean,
    includePaddings: Boolean,
    crop: {
        top: Number,
        left: Number,
        bottom: Number,
        right: Number
    },
}",
    }
  }
  description="Take screenshots of the tested page."
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/actions/take-screenshot.html"
/>
`;
